
From: NeilBrown <neilb@cse.unsw.edu.au>

From: Greg Banks <gnb@sgi.com>

NFS/TCP has been used widely for a long time now, and is now enabled by many
distros and typically recommended for lossy or congested networks.  This patch
removes the EXPERIMENTAL tag, updates the help text, and changes arch
defconfigs to set it by default (note that several already did so).

Signed-off-by: Neil Brown <neilb@cse.unsw.edu.au>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/alpha/defconfig |    2 +-
 25-akpm/arch/i386/defconfig  |    2 +-
 25-akpm/arch/ia64/defconfig  |    2 +-
 25-akpm/arch/ppc/defconfig   |    2 +-
 25-akpm/arch/s390/defconfig  |    2 +-
 25-akpm/fs/Kconfig           |   10 ++++++----
 6 files changed, 11 insertions(+), 9 deletions(-)

diff -puN arch/alpha/defconfig~knfsd-mark-nfs-tcp-server-not-experimental arch/alpha/defconfig
--- 25/arch/alpha/defconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.056506400 -0700
+++ 25-akpm/arch/alpha/defconfig	2004-06-23 22:12:01.067504728 -0700
@@ -791,7 +791,7 @@ CONFIG_NFS_V3=y
 CONFIG_NFSD=m
 CONFIG_NFSD_V3=y
 # CONFIG_NFSD_V4 is not set
-# CONFIG_NFSD_TCP is not set
+CONFIG_NFSD_TCP=y
 CONFIG_LOCKD=m
 CONFIG_LOCKD_V4=y
 CONFIG_EXPORTFS=m
diff -puN arch/i386/defconfig~knfsd-mark-nfs-tcp-server-not-experimental arch/i386/defconfig
--- 25/arch/i386/defconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.058506096 -0700
+++ 25-akpm/arch/i386/defconfig	2004-06-23 22:12:01.068504576 -0700
@@ -1148,7 +1148,7 @@ CONFIG_NFS_FS=y
 # CONFIG_NFS_DIRECTIO is not set
 CONFIG_NFSD=y
 # CONFIG_NFSD_V3 is not set
-# CONFIG_NFSD_TCP is not set
+CONFIG_NFSD_TCP=y
 CONFIG_LOCKD=y
 CONFIG_EXPORTFS=y
 CONFIG_SUNRPC=y
diff -puN arch/ia64/defconfig~knfsd-mark-nfs-tcp-server-not-experimental arch/ia64/defconfig
--- 25/arch/ia64/defconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.059505944 -0700
+++ 25-akpm/arch/ia64/defconfig	2004-06-23 22:12:01.069504424 -0700
@@ -999,7 +999,7 @@ CONFIG_NFS_DIRECTIO=y
 CONFIG_NFSD=y
 CONFIG_NFSD_V3=y
 # CONFIG_NFSD_V4 is not set
-# CONFIG_NFSD_TCP is not set
+CONFIG_NFSD_TCP=y
 CONFIG_LOCKD=y
 CONFIG_LOCKD_V4=y
 CONFIG_EXPORTFS=y
diff -puN arch/ppc/defconfig~knfsd-mark-nfs-tcp-server-not-experimental arch/ppc/defconfig
--- 25/arch/ppc/defconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.061505640 -0700
+++ 25-akpm/arch/ppc/defconfig	2004-06-23 22:12:01.070504272 -0700
@@ -1230,7 +1230,7 @@ CONFIG_NFS_V3=y
 CONFIG_NFSD=y
 CONFIG_NFSD_V3=y
 # CONFIG_NFSD_V4 is not set
-# CONFIG_NFSD_TCP is not set
+CONFIG_NFSD_TCP=y
 CONFIG_LOCKD=y
 CONFIG_LOCKD_V4=y
 CONFIG_EXPORTFS=y
diff -puN arch/s390/defconfig~knfsd-mark-nfs-tcp-server-not-experimental arch/s390/defconfig
--- 25/arch/s390/defconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.062505488 -0700
+++ 25-akpm/arch/s390/defconfig	2004-06-23 22:12:01.071504120 -0700
@@ -422,7 +422,7 @@ CONFIG_NFS_V3=y
 CONFIG_NFSD=y
 CONFIG_NFSD_V3=y
 # CONFIG_NFSD_V4 is not set
-# CONFIG_NFSD_TCP is not set
+CONFIG_NFSD_TCP=y
 CONFIG_LOCKD=y
 CONFIG_LOCKD_V4=y
 CONFIG_EXPORTFS=y
diff -puN fs/Kconfig~knfsd-mark-nfs-tcp-server-not-experimental fs/Kconfig
--- 25/fs/Kconfig~knfsd-mark-nfs-tcp-server-not-experimental	2004-06-23 22:12:01.064505184 -0700
+++ 25-akpm/fs/Kconfig	2004-06-23 22:12:01.076503360 -0700
@@ -1469,11 +1469,13 @@ config NFSD_V4
 	  If unsure, say N.
 
 config NFSD_TCP
-	bool "Provide NFS server over TCP support (EXPERIMENTAL)"
-	depends on NFSD && EXPERIMENTAL
+	bool "Provide NFS server over TCP support"
+	depends on NFSD
+	default y
 	help
-	  Enable NFS service over TCP connections.  This the officially
-	  still experimental, but seems to work well.
+	  If you want your NFS server to support TCP connections, say Y here.
+	  TCP connections usually perform better than the default UDP when
+	  the network is lossy or congested.  If unsure, say Y.
 
 config ROOT_NFS
 	bool "Root file system on NFS"
_
